### PR TITLE
Migrate to Kotlin 1.9 and AGP 8

### PR DESCRIPTION
**Pull request by kalinja**

### CODE REVIEW

**Review by smidf — CHANGES_REQUESTED** [review]

File: kaal-presentation/src/main/kotlin/cz/eman/kaal/presentation/util/DiffCallbacks.kt [hunk 1]
```diff
@@ -19,7 +19,7 @@ import androidx.recyclerview.widget.DiffUtil
  *
  * @since 0.8.0
  */
-class DiffCallback<in T>(
+class DiffCallback<in T: Any>(
```
[the remark this answers is not in the document]
Comment (kalinja) [reply kaal-presentation/src/main/kotlin/cz/eman/kaal/presentation/util/DiffCallbacks.kt:22]: Because the method `areItemsTheSame` is not allowing nullable items anymore. The definition is:
`public abstract boolean areItemsTheSame(@NonNull T oldItem, @NonNull T newItem);`
One option is this as I have done, the other is do not allow oldItems and newItems to be nullable as
```
    private val oldItems: List<T & Any>,
    private val newItems: List<T & Any>,
```
    
  I don't know which solution is better but one of these is must.

File: kaal-domain/build.gradle.kts [hunk 2]
```diff
@@ -17,12 +17,6 @@ dependencies {
     testImplementation(Dependencies.Test.kotlinTest)
 }
 
-tasks.withType<KotlinCompile> {
-    kotlinOptions {
-        jvmTarget = "1.8"
-    }
-}
-
```
Comment (smidf) [inline kaal-domain/build.gradle.kts:25]: Why did you remove this?

File: kaal-presentation/src/main/kotlin/cz/eman/kaal/presentation/util/DiffCallbacks.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (smidf) [reply kaal-presentation/src/main/kotlin/cz/eman/kaal/presentation/util/DiffCallbacks.kt:22]: OK 👍

File: kaal-domain/build.gradle.kts (hunk 2 above)
Comment (kalinja) [reply kaal-domain/build.gradle.kts:25]: Because it is not possible to compile it when we set jvmTarget to 1.8 here. I was following the same as it was done in maps

Comment (smidf) [reply kaal-domain/build.gradle.kts:25]: Apply changes from f8d4fda1c1f0ab9c8eb3f65e0c035782472765c5

Comment (kalinja) [reply kaal-domain/build.gradle.kts]: Ok, done

**Review by smidf — APPROVED** [review]
